### PR TITLE
Higher level planning

### PR DESCRIPTION
**Pull request by fosterlynn**
I branched from the last plan branch, so this includes both operational planning and higher level planning, processes, etc.  I'll close the last plan branch.

I will do some examples, which might cause some changes.  But I wanted to get this idea out there for review @elf-pavlik @bhaugen .

### CODE REVIEW

File: docs/introduction/estimates.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+* <b>Comparative Analysis</b>.  Often different plans will be created for the same basic data set.  One example is when doing risk analysis or other comparative analysis.  Different assumptions might scew a plan in different useful directions, for example a "normal scenario" and a "worst case scenario".
```
Comment (bhaugen) [inline docs/introduction/estimates.md]: "scew" s/b "skew"

File: docs/introduction/estimates.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+* <b>Network Analysis</b>.  This is an analytical look at all or some of the actual and/or potential resource flows for a scope, often a community or region.  This can be modeled using higher level types of processes and types of resources, and could include intents, commitments, and/or economic events.  One use of this kind of analysis is to identify gaps and opportunities to enhance the economic network closer to home for additional community resilience.
```
Comment (bhaugen) [inline docs/introduction/estimates.md]: "enhance the economic network closer to home for additional community resilience" is obscure. Maybe "keep resources circulating in a community to improve economic health and resilience"? Or something like that?

File: docs/introduction/plan.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# Operational Planning
+
+An operational plan is a schedule of related operational processes, that constitute a body of scheduled work with defined deliverable(s).  A plan normally contains one or more process resource flows, one for each deliverable.
+
+Processes in a plan can be part of different scopes, if the different scopes are tightly coordinated with pre-agreed rules, for example sub-organizations of a main organization, or working together in an ongoing tightly coordinated supply chain that doesn't require new decisions.  If not, or if the agents prefer, then requirements from another scope should become deliverables for that scope's plan.  Different batch sizes could trigger a new plan for inputs to the main deliverable too.  But all of this does not affect the vocabulary or model.  Plans can relate to each other through resource flows just like Processes.
```
Comment (bhaugen) [inline docs/introduction/plan.md]: This sentence was difficult to parse:
"Processes in a plan can be part of different scopes, if the different scopes are tightly coordinated with pre-agreed rules, for example sub-organizations of a main organization, or working together in an ongoing tightly coordinated supply chain that doesn't require new decisions."

Suggest dropping the last clause and simplifying a bit, like this:
"A plan can cover more than one scope, if the different scopes are tightly coordinated with pre-agreed rules, for example sub-organizations of a main organization, or an ongoing supply chain."

**Comment by fosterlynn** [comment]
I made some changes:  Separated Plan and Scenario (my new name for upper level estimates and summary actuals - I spent quite a bit of time on the thesaurus, but better name welcome!).  Added ScenarioDefinition as a knowledge layer object to define and maybe eventually configure types of scenarios.  Reflected this in the text docs.

Note: It looks like one project we are working with will need these concepts, mostly in the analytical side, both summary of past events and future intents.  They get going seriously in September.  We have some minimal material to work with now, and I'll take some time to see if this works with the data we have now.

**Comment by fosterlynn** [comment]
P.S. I'd also like to confirm in some detail that this works with your Intent use cases @elf-pavlik , per our last discussion.

File: docs/introduction/plan.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+A plan can cover more than one scope, if the different scopes are tightly coordinated with pre-agreed rules, for example sub-organizations of a main organization, or an ongoing supply chain.  If not, or if the agents prefer, then requirements from another scope should become deliverables for that scope's plan.  Different batch sizes could trigger a new plan for inputs to the main deliverable too.  But all of this does not affect the vocabulary or model.  Plans can relate to each other through resource flows just like Processes.
+
+Plans are used for understanding and coordinating what needs to happen for specific outputs. The size and complexity of a Plan is up to the people who are planning and coordinating the work.
+
+A Plan can be generated from a Recipe, or created without one.
+
+Some examples:
+
+* A communications group publishes posts for the larger group.  The communications group needs some of their posts translated into various languages.
```
Comment (bhaugen) [inline docs/introduction/plan.md]: Suggestion:
* A communications group creates articles for a larger group.  The communications group needs some of their articles to be translated into various languages. Both the creation of an article and its translation could be part of the same plan.

File: docs/introduction/plan.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+* An organization produces a standard batch size to stock, in anticipation of future orders.
+
+### Processes nested in a Plan
+
+When processes are "nested", it is not random, nor based on a taxonomy. It is based on what processes are actually part of the plan. And not all the inputs and outputs of nested processes are considered inputs and outputs of the plan, since some are both produced and consumed within the plan.  In the following simplified example, the flows between plant/weed and weed/harvest are not flows input or output of the plan.
+
+![process nesting](https://rawgit.com/valueflows/valueflows/master/release-doc-in-process/nesting.png)
+
+* There are some common situations for nested processes that will not be as simple as the above diagram.  Thes include:
+    * Action makes a difference.  When a piece of equipment or tool is "used", it is not gone at the end of the nesting process.  But if it is managed as a time-based resource with a calandar, some calendar duration is in fact consumed.  Or if a citable resource is created and then cited inside, it is also still there at the end of the nesting process.
```
Comment (bhaugen) [inline docs/introduction/plan.md]: Typo alert: "calandar, some calendar" s/b "calendar, some calendar"

File: docs/introduction/plan.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    * Batch or lot size makes a difference.  If a deliverable is say 5 of some asssembled item, and 20 of some input component are needed to maek that item, but the minimum batch size that makes sense is 100 of them, then there will be 80 left in inventory at the end of the nesting process, also an output.
```
Comment (bhaugen) [inline docs/introduction/plan.md]: Typo alert: asssembled -> assembled

File: docs/introduction/plan.md (hunk 5 above)
Comment (fosterlynn) [reply docs/introduction/plan.md]: fixed

File: docs/introduction/plan.md (hunk 4 above)
Comment (fosterlynn) [reply docs/introduction/plan.md]: changed

File: docs/introduction/plan.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+
+### Planning from a Recipe
+
+Plans can be generated from a recipe.  This is called a "demand explosion".  The generation might take into account current inventories, batch sizes, etc., so it is not necessarily a "pure" reflection of the recipe.  In addition, often plans are tweaked after generation from a recipe, depending on how firm and exact the recipe is.  A manufacturing recipe might be more exact than a recipe for a more general business process.  For these reasons, a plans is decoupled from the recipe that generated it in the vocabulary.  It maintains only the references to the resource and process specifications that were supplied from the recipe.
```
Comment (bhaugen) [inline docs/introduction/plan.md]: Suggested improvement to the first sentence:
Plans can be generated from a recipe by scaling the recipe according to the demanded quantity of the end outputs.

Comment (fosterlynn) [reply docs/introduction/plan.md]: done

**Pull request merged**